### ISSUE TITLE
Who is using router5?

### ISSUE THREAD

**Issue by troch** [router5/router5#161]
It is now more than two years router5 was first released. Never too late to ask that question: If you or your company are using router5 in production, let everyone know!

You can also provide me with a logo, I am in the middle of revamping the documentation site and will add a "Who is using router5" page or section.

**Comment by motet-a** [comment]
Hello,

I used it to get rid of react-router in a not-so-little SPA (with many popups and tricky history management on some pages). This app originally used react-router v3, I was not a huge fan of it and I had the overall feeling it could not easily work with react-router v4 (it is a totally breaking full rewrite). Our needs were quite tricky and specific, whereas IMHO react-router v4 seemed to be made for much simpler apps. So I discovered router5, and decided to replace react-router after some prototyping.

router5 did the job perfectly. It is probably more complex than react-router but it allows to do _anything_ you want with routes and the history. Thanks for this library :bowing_man:.

(Sorry, I think it is too soon to mention the app in the docs. I hope you won’t blame me for misusing this thread :upside_down_face:)

**Comment by kelkes** [comment]
we are using router5 in all of our react projects. (github.com/trigo-at)

**Comment by jtheisen** [comment]
Pretty sure I'm going to use it - I really just searched for something that does what your path-parser does, as I didn't think any full router was going to be flexible enough. Router5 apparently is.

I'm using it at VisBricks, here's a [logo](https://www.visbricks.com/img/visbricks-logo-120.png). It's not really consumer facing software though, so nothing to show to the public.

Regarding the docs I have a suggestion: I would make it clearer that and how the following two things are really separate components:

1. The router5 proper, used by implementing middlewares and plugins.
2. The "tree of routes", implemented by and used through the listeners plugin (right?) and registering a callback for each route on it.

The first can be used without the second and should arguably better understood first. (And I don't think all apps really have this kind of route hierarchy - it depends on how your user interface is designed.)

Thanks for this great piece of code!

**Comment by nhusher** [comment]
Using Router5 on the main [Faraday](https://faraday.io) application. It's not publicly available, though

**Comment by troch** [comment]
Subscribed, looking forward to watching it 😄

**Comment by aksel** [comment]
We use it @ [artboost.com](artboost.com). Isomorphic React, with Redux.

react-router v3 has a bug with session storage, that does not appear to be fixable. v4 does not support transition blocking, i.e. no awaiting async data, i.e. worse SEO stuff.

Took a couple of weeks, but I finished the transition today 😄 

Working with router5 has been a joy. It took a couple of days to wrap my mind around how it works, but after I understood it, it was all smooth sailing.

No open-source as of yet, but next week I'm going to make a repo with an example following our structure.

**Comment by cyberhck** [comment]
@troch we're using router5 but not launched yet, actually we're just starting and it's a huge app so it will take some time. However I AM looking forward to new documentation though, right now it has concepts, but I can't seem to find resources on how to use it in a proper way in production though :)

I just subscribed to @zacksiri I'm very much looking forward to know how he's using that on production :)

**Comment by troch** [comment]
> I can't seem to find resources on how to use it in a proper way in production though

There isn't a proper way, because it is un-opinionated ppl tend to come up with their own set-up. The documentation is very much how I use it, but I don't want to push my opinions onto people. Some users might have spent more time optimising and refining the way I use router5 than I have.

**Comment by cyberhck** [comment]
true, and I do like that router5 is opinionated, just thought maybe there's a way which is working well for you, specially the nested route, where only child route gets re-rendered.

**Comment by amok** [comment]
I have router5 in my list of "golden" js libs.
router5 is used by one leading VoIP US company I've been working at, on one of their huge projects.
I continue using router5 but also we are not launched yet - it is also a big project related to agriculture field.
@troch thank you

**Comment by cyberhck** [comment]
It's truly awesome, I used react-router, redux-little-router, and bunch of other things, and router5 is the best. There's one thing I like the most about and it's nearly impossible to do with any other router easily: promised based prevent navigation. Say we want to display a custom dialog to notify user if he's about to leave the route, (not talking about refresh, only the route change), router5 is the only with good api for that.

**Comment by aksel** [comment]
@cyberhck Regarding nested routes, I go through the routes recursively, and find the appropriate component to render. [Link to gist](https://gist.github.com/aksel/afb81f22ce44efc90a575470ad220a2d)

If you use Redux, you can use a selector, and map the component to props of the parent. You can also just use the function in the route node, but that quickly adds up to duplicated code. But by mapping it to props, it does mean the parent has to re-render on child change. But I believe it has to re-render anyhow. Maybe if you wrap the child in some sort of HOC, that would prevent i? I don't know.

And regarding preventing navigation while awaiting promise resolution **(and redirection on rejection!!!!)**, that is the _primary_ reason we migrated to router5. Great stuff!

**Comment by afarriaga** [comment]
Hi, thanks for this awesome router. Using router5 in my project also. I have a question around which package to use. My application uses `redux-observable` and `redux-router5` but I noted there is also `rxjs-router5`. Any hints/recommendations if it would be more suited to use `rxjs-router5` instead of hooking the router5 to the redux store?
I guess the benefit would be that then the routing would be handled by `redux-observable` as an epic. Not sure if this makes sense. :]

**Comment by troch** [comment]
Short answer is: you don't need `rxjs-router5` with `redux` and `redux-observable`, using it would imply not adding routing state to your redux store. It is designed (like `xstream-router5`) to be consumed in app which are purely reactive, like the ones you can create with `cycle.js`.

**Comment by afarriaga** [comment]
Thanks a lot for quick reply and for the clarification! 👍

**Comment by Aarbel** [comment]
www.clovis.pro uses it

**Comment by alextes** [comment]
We handle 200K+ visitors and millions of page views per day with router5. Can't give you a logo or name 😞 , but know your lib sees heavy use 😄 .

**Comment by cyberhck** [comment]
Hi there, not a relevant comment for this issue, but since you mentioned in this issue, I'm just curious. How's the new documentation going? You mentioned that you were working on it? I'd love to see how everyone is using this, even though I'm very happy with this and am using for all of my react project, I'd love to squeeze everything that might be useful from router5 before rolling custom workarounds :)

I AM working on a small starter boilerplate on golang and react (separate), so maybe someone can help me organize the routing better :slightly_smiling_face:

**Comment by zacksiri** [comment]
So Just an update on this thread.

+ my site https://www.codemy.net now is using router5 in production. 
+ [free episode](https://www.codemy.net/posts/react-intermediate-basic-links-and-routes-004) showing how i'm converting an old react router 3 app to using router5.
+ [template app](https://github.com/frontojs/template) With router5 setup out of the box.

**Comment by cyberhck** [comment]
actually I'm using router5 with SSR, it's working great, you can have a look at our boilerplate which is very basic and we've been meaning to update, but no time yet: github.com/crazyfactory/ts-react-boilerplate

**Comment by dogoku** [comment]
Used in a few projects in Singapore Exchange, including the revamped website: https://www2.sgx.com

We are not using any framework and only depending on webcomponents, so router5 was a great fit for us, as it provides routing, in a clean way with no other dependencies. Much love and respect from SGX.

**Comment by cweekly** [comment]
Router5 looks amazing. Incorporating it into my current project.

**Comment by FireworksX** [comment]
https://scores24.live/en uses router5

**Comment by Dragomir-Ivanov** [comment]
https://www.repairsetup.com uses router5
